src/parseDef.ts

```diff
@@ -30,253 +30,267 @@ import { JsonSchema7SetType, parseSetDef } from "./parsers/set";
-  | JsonSchema7NativeEnumType
-  | JsonSchema7NullType
-  | JsonSchema7NumberType
-  | JsonSchema7ObjectType
-  | JsonSchema7RecordType
-  | JsonSchema7TupleType
-  | JsonSchema7UnionType
-  | JsonSchema7UndefinedType
-  | JsonSchema7RefType
-  | JsonSchema7NeverType
-  | JsonSchema7MapType
-  | JsonSchema7AnyType
-  | JsonSchema7NullableType
-  | JsonSchema7AllOfType
-  | JsonSchema7UnknownType
-  | JsonSchema7SetType
-) & { default?: any; description?: string }
+export type JsonSchema7TypeUnion =
+    | JsonSchema7StringType
+    | JsonSchema7ArrayType
+    | JsonSchema7NumberType
+    | JsonSchema7BigintType
+    | JsonSchema7BooleanType
+    | JsonSchema7DateType
+    | JsonSchema7EnumType
+    | JsonSchema7LiteralType
+    | JsonSchema7NativeEnumType
+    | JsonSchema7NullType
+    | JsonSchema7NumberType
+    | JsonSchema7ObjectType
+    | JsonSchema7RecordType
+    | JsonSchema7TupleType
+    | JsonSchema7UnionType
+    | JsonSchema7UndefinedType
+    | JsonSchema7RefType
+    | JsonSchema7NeverType
+    | JsonSchema7MapType
+    | JsonSchema7AnyType
+    | JsonSchema7NullableType
+    | JsonSchema7AllOfType
+    | JsonSchema7UnknownType
+    | JsonSchema7SetType;
+
+export type JsonSchema7Type = JsonSchema7TypeUnion & {
+    default?: any;
+    description?: string;
+};
 
 export function parseDef(
-  def: ZodTypeDef,
-  refs: References,
-  preloadDefinitions?: {
-    definitions: Record<string, ZodSchema<any>>;
-    definitionsPath: string;
-    basePath: string[];
-  }
+    def: ZodTypeDef,
+    refs: References,
+    preloadDefinitions?: {
+        definitions: Record<string, ZodSchema<any>>;
+        definitionsPath: string;
+        basePath: string[];
+    }
 ): JsonSchema7Type | undefined {
-  const definitionSchemas = getPreloadedDefinitionSchemas(
-    refs,
-    preloadDefinitions
-  );
+    const definitionSchemas = getPreloadedDefinitionSchemas(
+        refs,
+        preloadDefinitions
+    );
 
-  const seenItem = refs.items.find((x) => Object.is(x.def, def));
+    const seenItem = refs.items.find((x) => Object.is(x.def, def));
 
-  if (seenItem) {
-    return select$refStrategy(seenItem, refs);
-  }
+    if (seenItem) {
+        return select$refStrategy(seenItem, refs);
+    }
 
-  const newItem: Item = { def, path: refs.currentPath, jsonSchema: undefined };
+    const newItem: Item = {
+        def,
+        path: refs.currentPath,
+        jsonSchema: undefined,
+    };
 
-  refs.items.push(newItem);
+    refs.items.push(newItem);
 
-  const jsonSchema = selectParser(def, (def as any).typeName, refs);
+    const jsonSchema = selectParser(def, (def as any).typeName, refs);
 
-  if (jsonSchema) {
-    addMeta(def, jsonSchema);
-    addDefinitionSchemas(
-      jsonSchema,
-      definitionSchemas,
-      preloadDefinitions?.definitionsPath
-    );
-  }
+    if (jsonSchema) {
+        addMeta(def, jsonSchema);
+        addDefinitionSchemas(
+            jsonSchema,
+            definitionSchemas,
```

Comment:
Do you mind reverting these style changes? They're a bit distracting

Reply:
will do!

Reply:
Just pushed a new commit without any purely styling discrepancies